### ISSUE TITLE
Test install.sh on Travis-CI

### ISSUE THREAD

**Issue by zbeekman** [sourceryinstitute/opencoarrays#80]
This may require patching (only on Travis-CI, on the fly) the `install_prerequisites/build` script to accommodate the different versions of GCC available on the Linux container workers and that available via Homebrew. Building GCC from source is not feasible at all, on Travis-CI, due to build time-out limitations, but I *think* building MPICH and CMake should be possible. Might be blocked by https://github.com/travis-ci/travis-ci/issues/5335 but I just thought of a possible work around.

**Comment by zbeekman** [comment]
I have an test of `install.sh` running on Travis-CI that includes building MPICH and CMake on both [Mac OS X](https://travis-ci.org/zbeekman/opencoarrays/jobs/101219695) and [Ubuntu Linux](https://travis-ci.org/zbeekman/opencoarrays/jobs/101219698) :tada:

Unfortunately, the one downside is that the build time went from just under 2 minutes (because MPICH and CMake were cached or installed from binary packages by a package manager) to 27 minutes. I'm definitely going to find this frustrating, and as a work around this is what I propose:

 1. Only test the full install including MPICH and CMake when `install.sh` has changed or something in `install_prerequisites` has changed, OR, the build is detected to be a release candidate. (More on this last point later.)
 2. The install.sh script can still be tested the rest of the time, but without triggering a lengthy build of MPICH and CMake.

**Comment by zbeekman** [comment]
Also, I think it's getting be time that I refactor `.travis.yml` to break out the build/install/etc. commands into separate scripts. Embedding shell script in YAML is only so readable, and is a bit of a PITA to work with.

**Comment by zbeekman** [comment]
I may break that last comment out into its own issue.

**Comment by zbeekman** [comment]
Closed by #110 

Still need to speed some things up and avoid the long builds if `install.sh` or `install_prerequisites/*` are un-touched by the Travis-CI commit range being tested.

**Comment by zbeekman** [comment]
Seems like the install.sh script can fail without raising and error, reopening until I figure this out.

**Comment by rouson** [comment]
@zbeekman I could probably save you a lot of time in identifying where and why that happens.  I'm free most of the rest of the day if you'd like to share screen and diagnose this together.

**Comment by zbeekman** [comment]
@rouson Thanks for the offer! I'm going to have some dinner with some friends, and then might try looking at this again. If you are signed in to Skype, I can send you a Skype IM when I'm back and looking at this. If you're free and online, great! If not, I'll try to hunt it down myself and we can discuss further tomorrow.

**Comment by rouson** [comment]
@zbeekman don't hesitate to call my mobile number if I'm not showing as available on Skype. There are plenty of times when I am able to talk, but don't have Skype open and Skype sometimes shows me as being unavailable for no apparent reason.

**Comment by zbeekman** [comment]
@rouson I had some computer troubles last night, so I didn't have a chance to get in touch with you. I'm taking a look at this now, though. I'll sign into Skype, and if I get stuck, I'll give you a call.

**Comment by zbeekman** [comment]
I think I've determined the source of the test problems:

 1. FTP is not on the path
 2. GFortran is 5.2 right now on Ubuntu, whereas the build.sh script requires 5.3
 3. The build.sh script fails silently

What this means for us:

 - I will try to see if it's possible to upgrade to 5.3 on travis-ci Ubuntu images
 - The install.sh tests pass with flying colors on OS X builds, so I don't think we need to wait to mint a release.

**Comment by zbeekman** [comment]
Opening new issue, regarding gfortran 5.2 vs 5.3 on ubuntu and closing this one.
